File: examples/java/operations/child-contexts/test-child-context.java
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;
import software.amazon.awssdk.services.lambda.model.OperationType;
import software.amazon.lambda.durable.model.ExecutionStatus;
import software.amazon.lambda.durable.testing.LocalDurableTestRunner;

class TestChildContext {

    @Test
    void childContextSucceeds() {
        var handler = new BasicChildContext();
        var runner = LocalDurableTestRunner.create(String.class, handler);

        var result = runner.runUntilComplete("order-1");

        assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());
        assertEquals("order-1:validated:charged", result.getResult(String.class));
    }

    @Test
    void recordsContextOperation() {
        var handler = new BasicChildContext();
        var runner = LocalDurableTestRunner.create(String.class, handler);

        var result = runner.runUntilComplete("order-1");

        var contextOps = result.getOperations().stream()
            .filter(op -> op.getType() == OperationType.CONTEXT)
            .toList();
        assertFalse(contextOps.isEmpty());
    }
}

File: examples/csharp/operations/child-contexts/test-child-context.cs
using System.Linq;
using Amazon.Lambda.DurableExecution.Testing;
using Xunit;

public class TestChildContext
{
    [Fact]
    public async Task ChildContextSucceeds()
    {
        await using var runner = new DurableTestRunner<OrderEvent, ChargedOrder>(
            new BasicChildContextExample().Workflow);

        TestResult<ChargedOrder> result = await runner.RunAsync(new OrderEvent("order-1"));

        Assert.True(result.IsSucceeded);
        Assert.Equal("order-1", result.Result!.OrderId);
        Assert.True(result.Result.Charged);
    }

    [Fact]
    public async Task RecordsContextOperation()
    {
        await using var runner = new DurableTestRunner<OrderEvent, ChargedOrder>(
            new BasicChildContextExample().Workflow);

        TestResult<ChargedOrder> result = await runner.RunAsync(new OrderEvent("order-1"));

        // The testing SDK records child context operations as CONTEXT-kind steps.
        var contextOps = result.Steps.Where(s => s.Kind == OperationKind.Context).ToList();
        Assert.NotEmpty(contextOps);
        Assert.Equal(OperationStatus.Succeeded, contextOps[0].Status);
    }
}
